pkg/kubelet/kubelet_pods.go

```diff
@@ -326,7 +326,19 @@ func ensureHostsFile(fileName, hostIP, hostName, hostDomainName string, hostAlia
 		hostsFileContent = managedHostsFileContent(hostIP, hostName, hostDomainName, hostAliases)
 	}
 
-	return ioutil.WriteFile(fileName, hostsFileContent, 0644)
+	err = ioutil.WriteFile(fileName, hostsFileContent, 0644)
+	if err != nil {
+		return err
+	}
+	fileInfo, err := os.Stat(fileName)
+	if err != nil {
+		return err
+	}
+	filePerms := fileInfo.Mode().Perm()
+	if filePerms != 0644 {
+		klog.Warningf("Failed to set 0644 permissions on %s, actual permissions: %o", fileName, filePerms)
```

Comment:
Nit: I think we try and have warnings start with a lowercase.

Could you also indicate that depending on the file permissions, this could cause non-root processes in the container to not be able to access `/etc/hosts` and thus not make network calls.

Reply:
> Nit: I think we try and have warnings start with a lowercase.

I'm curious about why did we try and have warnings start with a lowercase? Start log with uppercase and error with lowercase would be a better style IMO.

FYI: https://github.com/golang/go/wiki/CodeReviewComments#error-strings